rsocket-core/src/main/java/io/rsocket/util/ByteBufPayload.java

```diff
@@ -179,4 +123,70 @@ public static Payload create(Payload payload) {
         payload.sliceData().retain(),
         payload.hasMetadata() ? payload.sliceMetadata().retain() : null);
   }
+
+  @Override
+  public boolean hasMetadata() {
+    return metadata != null;
+  }
+
+  @Override
+  public ByteBuf sliceMetadata() {
+    return metadata == null ? Unpooled.EMPTY_BUFFER : metadata.slice();
```

Comment:
I wonder whether this implicit conversion from null to EMPTY_BUFFER could be confusing for the null/"" edge case